google-play.php

```diff
@@ -125,6 +125,11 @@ public function parseApplication($packageName, $lang='en_US', $loc='US') {
     } else {
       $values["images"] = null;
     }
+    
+    // remove width & height
+    // remove -rw to get png type
+    $values["icon"] = preg_replace('/(\=[a-z0-9-]+)$/i', '=s280', $values["icon"]);
```

Comment:
Could you please explain what the `=s280` stands for, which you put in instead of the icon width/height?

Reply:
=s280  mean width=280 & height=280
i put it because icon full image is so big 

check
https://play-lh.googleusercontent.com/bYtqbOcTYOlgc6gqZ2rwb8lptHuwlNE75zYJu6Bn076-hTmvd96HH-6v7S0YUAAJXoJN=s180

https://play-lh.googleusercontent.com/bYtqbOcTYOlgc6gqZ2rwb8lptHuwlNE75zYJu6Bn076-hTmvd96HH-6v7S0YUAAJXoJN

Reply:
I think that's a bit mixing up then (it's not "remove w+h" but "replace"). You could always replace the size afterwards in your code. I'd rather leave it here as-is (might be a good indication on what "upstream" considers recommended values, whatever that will mean) than fixing it to some "random" value. Custom sizing shouldn't be done in a general-use library. The idea here is to provide you the "upstream data"; custom actions should then go to custom code. I tend to the same for removing size-specs from other images, too. (if it matters: that's how I use this lib; I replace size values in my code instad of here in the global place)

@BaseMax you will have the last word here of course :wink:

Reply:
Yes, you are right, but we must delete the width and length values to bring the original links to the images, not of specific sizes, and I put the code as an example
Secondly, deleting the -rw brings image links in .png format because the original links are in a .webp format and these images do not work on all browsers

Reply:
Pending Max' opinion: on the `-rw` removal I'd agree, `.webp` images are not a good "general standard" (I must admit I have problems with those as well when I encounter them – though I cannot remember a single case where I got them from this). Do you have an example for such an app?

Reply:
open this url on chrome & firefox 
https://play.google.com/store/apps/details?id=com.zhiliaoapp.musically.go 
and you'll see on firefox -rw was removed on a icon link and screenshots

Reply:
I see no `-rw` present for any of the image URLs retrieved with the current version of this library:

```php
    [icon] => https://play-lh.googleusercontent.com/LLUZLIcA7MbM5yLwAA-oTAt3q5kdDjIIfrEqR2mGLAxvVJcwruxJYQChwPDahLvSjFc=s180
    [featureGraphic] => https://play-lh.googleusercontent.com/SoKYaGxRqfpw7Xh4isCrb7tHjxl7w7g8S822stjT_ACMy025ax0zSXCSFVeq5HVNQ7k
    [images] => Array
        (
            [0] => https://play-lh.googleusercontent.com/nCeqIFSsd7MR92UCsVtTxQE-6LCU6koeM3g07cTy1BsL7IGnMSPydm8wo0k8iLEnPmwV=w720-h310
            [1] => https://play-lh.googleusercontent.com/lEqBUSyW-vC3dNkIizFPyc26gQUt1-TThzwR-rY5uV_9aHHNTYBzvw0YOKLnzLAuMt0=w720-h310
            [2] => https://play-lh.googleusercontent.com/0fHcPf2oQRLD29JREeg9F1oxfRsINylWWBktQ9nDDt4GuJIwvr6zJNbS36WEgi28PlY=w720-h310
        )
```

so for this one, no change is needed. As you mention Chrome and Firefox with the latter having it removed, maybe that's a question of the UA specified? Then, instead of rewriting the links, we might simply allow specifying the UA to be used with `stream_context_create()`. Currently, none is specified – which seems to work fine here on Linux. Not using any other OSs, I cannot speak for them; but if that makes a difference, I'd think it would be preferable that way. "Personal preferences" of "URL rewrites" shouldn't go into a core lib.

Reply:
Thanks for following up Izzy; So aren't this PR essentials and needed?

Best

Reply:
That's up for you to decide, @BaseMax – IMHO this should be a "base API" and thus focus on essentials. The class can always be "extended" if one needs "special stuff".

In this example, the API provides the full URLs as Google specifies them so everyone can freely process them. One wants the default height/width as specified by Google, another wants sizes stripped, and yet another wants specific sizes – you cannot make it "right" for everyone. As it currently is, all 3 parties can satisfy their needs by modifying returned URLs. If we remove sizing here at the base, we already exclude the user who wants the "Google defaults".

Also see [my second comment](#discussion_r550813678) above, explaining these details.